src/Symfony/Component/OptionsResolver/Tests/OptionsResolver2Dot6Test.php

```diff
@@ -499,27 +499,29 @@ public function testFailIfSetAllowedTypesFromLazyOption()
     }
 
     /**
+     * @dataProvider provideInvalidTypes
      * @expectedException \Symfony\Component\OptionsResolver\Exception\InvalidOptionsException
-     * @expectedExceptionMessage The option "foo" with value 42 is expected to be of type "string", but is of type "integer".
      */
-    public function testResolveFailsIfInvalidType()
+    public function testResolveFailsIfInvalidType($actualType, $allowedType, $exceptionMessage)
     {
-        $this->resolver->setDefined('foo');
-        $this->resolver->setAllowedTypes('foo', 'string');
+        $this->setExpectedException('InvalidArgumentException', $exceptionMessage);
```

Comment:
`setExpectedException` must be called between setAllowedTypes and resolve as the exception is to be expected then.